File: Examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/db/entity/StructuredMessage.java
package me.connect.sdk.java.sample.db.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

import me.connect.sdk.java.message.StructuredMessageHolder.Response;

@Entity
public class StructuredMessage {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @ColumnInfo(name = "connection_id")
    public int connectionId;

    @ColumnInfo(name = "serialized")
    public String serialized;

    @ColumnInfo(name = "entry_id")
    public String entryId;

    @ColumnInfo(name = "message_id")
    public String messageId;

    @ColumnInfo(name = "question_text")
    public String questionText;

    @ColumnInfo(name = "question_detail")
    public String questionDetail;

    @ColumnInfo(name = "answers")
    public List<Response> answers;

    @ColumnInfo(name = "selected_answer")
    public String selectedAnswer;

    @ColumnInfo(name = "type")
    public String type;
}

File: Examples/android/CMeSdkJava/app_rework_kt/src/main/kotlin/me/connect/sdk/java/samplekt/db/entity/StructuredMessage.kt
package me.connect.sdk.java.samplekt.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity

import androidx.room.PrimaryKey
import me.connect.sdk.java.message.StructuredMessageHolder.Response


@Entity
data class StructuredMessage(
        @PrimaryKey(autoGenerate = true)
        val id: Int = 0,

        @ColumnInfo(name = "connection_id")
        var connectionId: Int,

        @ColumnInfo(name = "serialized")
        var serialized: String,

        @ColumnInfo(name = "entry_id")
        var entryId: String,

        @ColumnInfo(name = "message_id")
        var messageId: String,

        @ColumnInfo(name = "question_text")
        var questionText: String,

        @ColumnInfo(name = "question_detail")
        var questionDetail: String,

        @ColumnInfo(name = "answers")
        var answers: List<Response>,

        @ColumnInfo(name = "selected_answer")
        var selectedAnswer: String? = null
)